common/src/main/scala/org/apache/celeborn/common/authentication/Credential.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.authentication
+
+/**
+ * The credential object that is passed to the token authentication provider.
+ *
+ * @param token The token that is used for authentication
+ * @param props The properties that associated with the token
+ */
+case class Credential(token: String, props: Map[String, String] = Map.empty)
```

Comment:
do you mean this? @pan3793

Reply:
almost, but maybe
```
case class PasswordCredential(username: String, password: String)
case class TokenCredential(token: String, properties: Map[String, String])
``

Reply:
done

Reply:
second thought, check client ip is also applicable for password authN cases, and user may want to have more custom rules, we should make the API likes

```
trait PasswordCredential {
  def username: String
  def password: String
  def extraInfo: Map[String, String] = Map.empty
}

case class DefaultPasswordCredential(
  username: String,
  password: String,
  extraInfo: Map[String, String] = Map.empty) implement PasswordCredential
```